### ISSUE TITLE
How to use the model to make prediction?

### ISSUE THREAD

**Issue by syu-look** [ZhaoningYu1996/HM-GNN#5]
Hi, thanks for your work. i trained the model with my own data. But I don't know how make prediction? you wrote evaluate function I saw, but i don't know the meaning of some parameters. now i have a list of smiles ,such as x = ["CCCC",...] y = [1,...] ,can you tell me how to predict?

**Comment by ZhaoningYu1996** [comment]
Hi,

Thank you for your interest in our work.
We did not implement a SMILES version preprocessing.
However, you can convert SMILES into .mol object first, then convert mol into networkx object.
You can refer to the code below as an example.
https://gist.github.com/fangkuoyu/dc785218e5d4d94c752e80f1aaba4fad
Then, you can use our evaluate function to make the prediction.
Hope this help.

Best

**Comment by syu-look** [comment]
sorry, because I am a beginner. I know to use the evaluation function  to predict. actually I don't know how to write the code to call it and what parameters means? can you describe it in detail?

**Comment by ZhaoningYu1996** [comment]
Hi, 

Do you know how to train your model using your dataset?
If you understand how to train, the testing is the same but need to add two lines:
model.eval()
with torch.no_grad():
If you are not familiar with training, we train the heterogeneous graph in a semi-supervised way. We mask testing nodes in the heterogeneous graph and update the model by CrossEntropyLoss. Then, we test the model by masking training nodes. You can refer to train_and_evaluate() in main.py. 

Hope this help,
Zhaoning

**Comment by syu-look** [comment]
Ok, I understand a little and  I am trying to write it now, Thanks for your reply.
Best.
